File: assets/ipl/solutions/19sp/lectures/2.18.19/Frog.cpp
#include<bits/stdc++.h>
using namespace std;
long long int dp[1000005];
int a[1000005];
int main(){
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  int n;
  cin>>n;
  for(int i=1;i<=n;i++){
    cin>>a[i];
    dp[i]=10000000000000LL;
  }
  dp[1]=0;
  for(int i=1;i<=n;i++){
    dp[i+2]=min(dp[i]+abs(a[i+2]-a[i]),dp[i+2]);
    dp[i+1]=min(dp[i]+abs(a[i+1]-a[i]),dp[i+1]);
  }
  cout<<dp[n]<<endl;
}

File: assets/ipl/solutions/19sp/lectures/2.18.19/Frog.py
n = int(input())
h = list(map(int, input().split()))
dist = [10**9] * n
dist[0] = 0
for i in range(0, n):
	if i + 1 < n:
		dist[i + 1] = min(dist[i + 1], dist[i] + abs(h[i + 1] - h[i]))
	if i + 2 < n:
		dist[i + 2] = min(dist[i + 2], dist[i] + abs(h[i + 2] - h[i]))
print(dist[n - 1])
